chore(release): add unreleased changelog entries (#547)
exoscale/terraform-provider-exoscale: CHANGELOG.md

IMPROVEMENTS:

- Removed deprecated and unused `delay` provider attribute #539

> - test: local-creds helper for acceptance runs #545

Reviewer: I'm not sure whether it makes sense to mention any dev or test-related changes. This is why I didn't include #533 here.

My rationale is that the changelog should mention user-facing changes, and ideally be directed towards the user (there's an argument to be made that commit messages shouldn't be used in a changelog directly for this reason either). So any changes that don't strictly impact the user are just noise from the perspective of a user IMO.

But there is no universal rule for this, so happy to hear other thoughts.

Author: Agreed on #545, will drop, given how things are done today.

On the rewording, I'd keep commit-style for now since it matches the 0.69.2 block above (e.g. sks: add Kubelet Image GC..., Fix SG resource schema version).

I'd rather do it as one pass over the whole CHANGELOG + CI/CD than piecemeal per PR. 

Will create a note so we can discuss it internally.

BUG FIXES:

- Block storage: handle BS volume deletion in instance update #546
- fix(compute_instance): suppress user_data diffs for pre-encoded input #518
- fix(security-group-rule): allow portless protocols (ESP, AH, GRE, IPIP) #531

> - fix(template): fall back to private visibility when name lookup returns not found #532

Reviewer: I would drop the type since it's already clear from the section, and make the scope a bit more user friendly, so:

```suggestion
- Instance: suppress user_data diffs for pre-encoded input #518
- Security group rule: allow portless protocols (ESP, AH, GRE, IPIP) #531
- Template: fall back to private visibility when name lookup returns not found #532
```

Although it would also make sense to use the suffix from the actual TF resource name, so:
```suggestion
- `compute_instance`: suppress user_data diffs for pre-encoded input #518
- `security_group_rule`: allow portless protocols (ESP, AH, GRE, IPIP) #531
- `template`: fall back to private visibility when name lookup returns not found #532
```

... could also work.

TBH we don't follow a strict convention about this, but maybe we should. WDYT?

BUG FIXES:

> - Block storage: handle BS volume deletion in instance update #546

Reviewer: Unrelated to your change, but so that we're at least consistent within a single release:

```suggestion
- `block_storage_volume`: handle BS volume deletion in instance update #546
```

Author: Ta :bowing_woman:

- `compute_instance`: suppress user_data diffs for pre-encoded input #518
- `security_group_rule`: allow portless protocols (ESP, AH, GRE, IPIP) #531
- `template`: fall back to private visibility when name lookup returns not found #532

DEPENDENCIES:

- upgraded `golang/x/net` and related dependencies #540

FEATURES:

> - feat(dbaas): add exoscale_dbaas_valkey_user resource #528

Reviewer: ```suggestion
- `dbaas`: add exoscale_dbaas_valkey_user resource #528
```

Author: Ta :bowing_woman:
